boopmain.ino

```diff
@@ -29,22 +29,37 @@ uint32_t wait_update = 200; // This is the number of MS between frames. e.g. 500
 uint32_t last_buttoncheck = 0;
 uint32_t wait_buttoncheck = 50;
 
+bool last_buttonpress = false; // This variable stores whether the button was pressed in the last loop
+
 ///////////////////////////////////////
 // Light Variables and Animation Variables
 ///////////////////////////////////////
 
 bool is_button_pressed = false; //Check for button press
-bool ledsON = false;
+bool leds_on = false;
 
 //List of vibe colours available
 uint32_t rainbowColors[] = {
-  0xFF0000, // Red
-  0xFF7F00, // Orange
-  0xFFFF00, // Yellow
-  0x00FF00, // Green
-  0x0000FF, // Blue
-  0x4B0082, // Indigo
-  0x8B00FF  // Violet
+0xFF0000,   // Red
+0xFF7F00,   // Orange
+0xFFFF00,   // Yellow
+0x00FF00,   // Green
+0x00FFFF,   // Cyan
+0x0000FF,   // Blue
+0x8A2BE2,   // Indigo
+0xFF00FF,   // Magenta
+0xFF1493,   // Pink
+0x00FF7F,   // Bright Green
+0xFF4500,   // Bright Orange
+0xFFFF66,   // Bright Yellow
+0x00FF00,   // Bright Lime
+0x00FFFF,   // Bright Aqua
+0x00BFFF,   // Bright Sky Blue
+0x9932CC,   // Bright Violet
+0xFF69B4,   // Bright Rose
+0xFF1493,   // Bright Hot Pink
+0x1E90FF,   // Electric Blue
+0xFF4500    // Electric Orange
```

Comment:
Great use of arrays.

another way which might be simpler is to use CHSV where hue is updated.

```
uint8_t hue = 0;
...
hue = hue + 30;
...
leds[i]  = CHSV(hue,255,255);

```

This will give a rainbow effect as well with less to manage. since hue is a `uint8_t` the max value is 255, and adding numbers above that will wrap around 255+2 = 1